services/provider/server/server.go

```diff
@@ -870,17 +879,29 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
 
-	desiredClientConfigHash := getDesiredClientConfigHash(channelName, storageConsumer)
+	storageCluster, err := s.getStorageCluster(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	inTransitEncryptionEnabled := false
+	if storageCluster.Spec.Network != nil && storageCluster.Spec.Network.Connections != nil &&
+		storageCluster.Spec.Network.Connections.Encryption != nil && storageCluster.Spec.Network.Connections.Encryption.Enabled {
+		inTransitEncryptionEnabled = true
+	}
```

Comment:
This pattern repeats itself. I would suggest to replace the `getStorageCluster` utils with a `IsEncryptionInTransitEnabled()` function and call it in both places

Reply:
Kept the getStorageCluster func as it's being used at another place too, created a separate isEncryptionInTransitEnabled() taking only the networkSpec.